snuba/api.py

```diff
@@ -308,37 +314,34 @@ def format_query(dataset, body, table, prewhere_conditions, final) -> str:
 
 
 @split_query
-def parse_and_run_query(dataset, body, timer):
+def parse_and_run_query(dataset, request: Request, timer):
     max_days, date_align = state.get_configs([
         ('max_days', None),
         ('date_align_seconds', 1),
     ])
 
-    to_date = util.parse_datetime(body['to_date'], date_align)
-    from_date = util.parse_datetime(body['from_date'], date_align)
+    to_date = util.parse_datetime(request.extensions['timeseries']['to_date'], date_align)
+    from_date = util.parse_datetime(request.extensions['timeseries']['from_date'], date_align)
     assert from_date <= to_date
 
     if max_days is not None and (to_date - from_date).days > max_days:
         from_date = to_date - timedelta(days=max_days)
 
-    # XXX: This should already be enforced by the schema at this point.
-    body.setdefault('conditions', [])
-
-    body['conditions'].extend([
+    request.query['conditions'].extend([
```

Comment:
Where is this logic that writes on the query going in to go long term (valid for line 335 342 etc as well)? 
is this going to stay here or move to the Request class itself?